### PR TITLE
feat: modify docs for Apache APISIX 2.1

### PR DESCRIPTION
**Pull request by gxthrj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,30 @@
 **Docker images are not official ASF releases but provided for convenience. Recommended usage is always to build the source.**
 
-## QuickStart
+## How To Build Image
+
+### Build an image from source
+
+**Docker images are not official ASF releases but provided for convenience. Recommended usage is always to build the source.**
```
Comment (moonming) [inline README.md:7]: repeat

**Review by moonming — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gxthrj) [reply README.md]: fixed

**Comment by gxthrj** [comment]
> NO, you can not delete this line

Keep one in the header of README

File: example/apisix_conf/config.yaml [hunk 2]
```diff
@@ -1,137 +1,22 @@
 apisix:
```
Comment (membphis) [inline example/apisix_conf/config.yaml:1]: missing Apache License header, any way we can fix this in new PR

File: example/apisix_conf/config.yaml [hunk 3]
```diff
@@ -1,137 +1,22 @@
...
-      - 127.0.0.1
-      - 'unix:'
-    #lua_shared_dicts:              # add custom shared cache to nginx.conf
-    #  ipc_shared_dict: 100m        # custom shared cache, format: `cache-key: cache-size`
 
 etcd:
   host:                           # it's possible to define multiple etcd hosts addresses of the same etcd cluster.
     - "http://172.18.5.10:2379"     # multiple etcd address
   prefix: "/apisix"               # apisix configurations prefix
   timeout: 3                      # 3 seconds
```
Comment (membphis) [inline example/apisix_conf/config.yaml:21]: the default value is  30 sec now

File: example/apisix_conf/config.yaml [hunk 4]
```diff
@@ -1,137 +1,22 @@
 apisix:
   node_listen: 9080              # APISIX listening port
-  enable_heartbeat: true
-  enable_admin: true
-  enable_admin_cors: true         # Admin API support CORS response headers.
-  enable_debug: false
-  enable_dev_mode: false          # Sets nginx worker_processes to 1 if set to true
-  enable_reuseport: true          # Enable nginx SO_REUSEPORT switch if set to true.
-  enable_ipv6: true
```
Comment (membphis) [inline example/apisix_conf/config.yaml:9]: do we need to disable `ipv6 ` by default for docker?

Comment (gxthrj) [reply example/apisix_conf/config.yaml:9]: Done, we'd better disable it.

File: example/apisix_conf/config.yaml (hunk 3 above)
Comment (gxthrj) [reply example/apisix_conf/config.yaml]: Done.

File: example/apisix_conf/config.yaml (hunk 2 above)
Comment (gxthrj) [reply example/apisix_conf/config.yaml:1]: We can fix this in new PR, so many files do miss Apache License header.

Comment (membphis) [reply example/apisix_conf/config.yaml:1]: I do not find any related issue, so I created one: https://github.com/apache/apisix-docker/issues/92

**Review by membphis — APPROVED** [review]
LGTM

**Comment by membphis** [comment]
@gxthrj Is this PR related to any issue? Please confirm

**Review by moonming — APPROVED** [review]

**Comment by gxthrj** [comment]
> @gxthrj Is this PR related to any issue? Please confirm

We can close #87

**Pull request merged**